cmd/xk6-depsync/main.go

```diff
@@ -0,0 +1,112 @@
+// main is the main package for xk6-depsync, a script that checks and provides commands to synchronize
+// common dependencies with k6 core.
+package main
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
+)
+
+const (
+	k6Core     = "go.k6.io/k6"
+	k6GoModURL = "https://proxy.golang.org/" + k6Core + "/@v/%s.mod"
+)
+
+func main() {
+	gomod := "go.mod"
+	if len(os.Args) >= 2 {
+		gomod = os.Args[1]
+	}
+
+	file, err := os.Open(gomod)
+	if err != nil {
+		log.Fatalf("opening local go.mod: %v", err)
+	}
+
+	ownDeps, err := dependencies(file)
```

Comment:
Consider using https://pkg.go.dev/golang.org/x/mod@v0.13.0/modfile#Parse instead of custom parsing logic, it would be a more robust solution....

Reply:
I've given this a swing in the latest two commits. I wouldn't say I'm _against_ it, but I'm not very convinced either. I don't see the code significantly less complex or significantly shorter either.

LMK what you think about this. If you prefer it this way, I'm happy letting it stay, otherwise I can drop the commits and get back to the original implementation. Your call :)

Reply:
I didn't recommend it because of the complexity or length of the code. If there is a well-maintained, high-quality library for reading a file format, it is advisable to use it. This way you can avoid that your own parser does not exactly follow the file format specification.

BTW this is just a suggestion, it's up to the code maintainer to decide :)
( @olegbespalov ? )